File: PHP/test/Kata/AccountNumberTest.php
<?php
namespace Test\Kata;

use \Kata\AccountNumber;

class AccountNumberTest extends \PHPUnit_Framework_TestCase
{

    /** @test */
    public function shouldBeEqualInList()
    {
        $accountNumbers = [new AccountNumber("123456789")];

        $this->assertEquals(1, count($accountNumbers));
        $this->assertEquals(new AccountNumber("123456789"), $accountNumbers[0]);
    }

    /** @test */
    public function shouldNotBeEqual()
    {
        $this->assertNotEquals(new AccountNumber("123456789"), new AccountNumber("123456788"));
    }

    /** @test */
    public function shouldListNumberAsString()
    {
        $accountNumber = new AccountNumber("123456789");

        $this->assertEquals("123456789", "" . $accountNumber);
    }

    /**
     * @test
     * @expectedException \InvalidArgumentException
     */
    public function shouldValidateNumberLength()
    {
        // $this->expectException(\InvalidArgumentException::class);
        new AccountNumber("12345678");
    }

    /**
     * @test
     * @expectedException \InvalidArgumentException
     */
    public function shouldValidateNumber()
    {
        // $this->expectException(\InvalidArgumentException::class);
        new AccountNumber("12345678n");
    }

}

File: Java/src/test/java/kata/AccountNumberTest.java
package kata;

import static org.junit.Assert.assertEquals;
import static org.junit.Assert.assertNotEquals;

import java.util.Arrays;
import java.util.List;

import org.junit.Test;

public class AccountNumberTest {

    @Test
    public void shouldBeEqualInList() {
        List<AccountNumber> accountNumbers = Arrays.asList(new AccountNumber("123456789"));

        assertEquals(1, accountNumbers.size());
        assertEquals(new AccountNumber("123456789"), accountNumbers.get(0));
    }

    @Test
    public void shouldNotBeEqual() {
        assertNotEquals(new AccountNumber("123456789"), new AccountNumber("123456788"));
    }

    @Test
    public void shouldListNumberAsString() {
        AccountNumber accountNumber = new AccountNumber("123456789");

        assertEquals("123456789", accountNumber.toString());
    }

    @Test(expected = IllegalArgumentException.class)
    public void shouldValidateNumberLength() {
        new AccountNumber("12345678");
    }

    @Test(expected = IllegalArgumentException.class)
    public void shouldValidateNumber() {
        new AccountNumber("12345678n");
    }

}
